tests/parse_metadata.py

```diff
@@ -151,7 +154,7 @@ def read_metadata(distribution: str) -> StubMetadata:
     given in the `requires` field, for example.
     """
     with Path("stubs", distribution, "METADATA.toml").open("rb") as f:
-        data: dict[str, object] = tomli.load(f)
+        data: dict[str, Any] = tomli.load(f)
```

Comment:
Why is this change required? I deliberately used `dict[str, object]` when I wrote this module so that we'd have very strict type checking in this function. This makes the code much less type-safe.

Reply:
If necessary, I'd prefer a `cast()` here if it lets us avoid `Any` for the value type:

```suggestion
        data = cast(dict[str, object], tomli.load(f))
```

Reply:
```
E:\Users\Avasam\Documents\Git\typeshed\tests\parse_metadata.py
  E:\Users\Avasam\Documents\Git\typeshed\tests\parse_metadata.py:199:29 - error: Type of "get" is partially unknown
    Type of "get" is "Overload[(__key: Unknown, /) -> (Unknown | None), (__key: Unknown, __default: Unknown | _T@get, /) -> (Unknown | _T@get)]" (reportUnknownMemberType)
  E:\Users\Avasam\Documents\Git\typeshed\tests\parse_metadata.py:199:29 - error: Type of "get" is partially unknown
    Type of "get" is "Overload[(__key: Unknown, /) -> (Unknown | None), (__key: Unknown, __default: Unknown | _T@get, /) -> (Unknown | _T@get)] | Overload[(__key: str, /) -> (set[str] | None), (__key: str, 
__default: set[str] | _T@get, /) -> (set[str] | _T@get)]" (reportUnknownMemberType)
  E:\Users\Avasam\Documents\Git\typeshed\tests\parse_metadata.py:12:20 - error: Import "Any" is not accessed (reportUnusedImport)
4 errors, 0 warnings, 0 informations
```

Basically at line 195 `tools_settings` is `object | dict[str, set[str]]` and `isinstance(tools_settings, dict)` narrows `object` to `dict[Unknown, Unknown]`.
So the type becomes `dict[Unknown, Unknown] | dict[str, set[str]]`

In this case a cast doesn't help.

I could fix it with a typeddict like I did in check_consistent. Or a custom type-guard. Or add `reportUnknownMemberType` witht eh suppressions up to, but I don't like that I had to add those in the first place.

Reply:
I'd _really_ rather not introduce `Any` here; it'll mean mypy will verify much less of this function and make the code much less type-safe imo. So I guess I'd advocate adding `reportUnknownMembeType=false` to the suppressions at the top of the file.

Reply:
I'll do that. I'm fine with it because, as the comment says. The whole point of this file is to abstract away those errors. So we expect a bunch of `Unknown` and partially `Unknown` because we're checking the content of files that have not been validated yet.